### ISSUE TITLE
Create a maven BOM pom module for AndHow

### ISSUE THREAD

**Issue by eeverman** [eeverman/andhow#529]
AndHow is currently distributed as a 'fat' jar created from the contents of two jars: `anyhow-core.jar` and `andhow-annotation-processor.jar`, both of which are created in this project.  

This is really just to simplify things so AndHow can be included as a single Maven dependency, however, if you want to create the smallest deployable artifact possible for an application, you can do better.  The actual requirement for AndHow in an application is this:
* `andhow-annotation-processor.jar` This annotation processor is required at compile time **_only_**.  It is not required at runtime so it does not need to be deployed with the application.
* `anyhow-core.jar` Required at compile time and at runtime.

Thus, a BOM style pom file with a dependencyManagement section that includes `andhow-annotation-processor.jar` as `provided` scope and `anyhow-core.jar` as `compile` scope would provide an easy path for users who want to create the smallest possible app.

To complete this:
* Create a new module called `andhow-bom` with a pom file of package type `pom`.  It should have a dependency management section as described above.
* Update[ this mini-app's pom file](https://github.com/eeverman/andhow/blob/main/andhow-testing/andhow-simulated-app-tests/example-app-2/pom.xml) to import and use the BOM dependency.  This will serve as both a test and an example.

**Comment by alex-kar** [comment]
@eeverman I would like to take it, but I'm not sure I understand the idea.
If we create new module `andow-bom` with `dependencyManagment` section and import it in `example-app-2` as
```
<dependencyManagement>
	<dependencies>
		<dependency>
			<groupId>${project.groupId}</groupId>
			<artifactId>andhow-bom</artifactId>
			<version>${project.version}</version>
			<type>pom</type>
			<scope>import</scope>
		</dependency>
	</dependencies>
</dependencyManagement>
```
Then we anyway should reference those dependencies `andhow-annotation-processor` and `anyhow-core` in `dependencies` section, otherwise they won't be include.

**Comment by eeverman** [comment]
@alex-kar Right, its low priority.  This is really intended for people who want to create as small a jar as possible - Probably because they want a lambda to startup quickly.  Only 'core' is needed at runtime - you can leave out the annotation-processor.  But, to do that, you have to know what you are doing and get the scopes of the dependencies correct.  This was intended to simplify that because the dependency management will bring in the scopes.

This would require some testing to make sure it works as expected (I didn't test this out, but this seems like the way BOMs are supposed to work - I'm not sure if the provided scope adds any wrinkles).

I'll pull the 'help wanted tag' off and assign to you - feel free to ignore for as long as you want.
